enigma-swing/src/main/java/cuchaz/enigma/gui/Gui.java

```diff
@@ -295,11 +295,15 @@ public void mousePressed(MouseEvent e) {
 			@Override
 			public void mousePressed(MouseEvent e) {
 				if (SwingUtilities.isRightMouseButton(e)) {
+					deobfPanel.deobfClasses.setSelectionRow(deobfPanel.deobfClasses.getClosestRowForLocation(e.getX(), e.getY()));
 					int i = deobfPanel.deobfClasses.getRowForPath(deobfPanel.deobfClasses.getSelectionPath());
 					if (i != -1) {
 						deobfPanelPopupMenu.show(deobfPanel.deobfClasses, e.getX(), e.getY());
 					}
 				}
+
+				// Only enable rename class if selected path is a class
+				deobfPanelPopupMenu.getRenameClass().setEnabled(deobfPanel.deobfClasses.getSelectedClass() != null);
```

Comment:
Shouldn't this happen in DeobfPanelPopupMenu.show?

Reply:
Yes you're right, changed